spec/models/cloud_volume/cloud_volume_operations_spec.rb

```diff
@@ -0,0 +1,46 @@
+RSpec.describe 'CloudVolume::Operations' do
+  let(:disk)         { FactoryBot.create(:disk) }
+  let(:ems)          { FactoryBot.create(:ems_vmware) }
+  let(:cloud_volume) { FactoryBot.create(:cloud_volume, :ext_management_system => ems) }
+  let(:user)         { FactoryBot.create(:user, :userid => 'test') }
+
+  context "queued methods" do
+    it "queues an attach task with attach_volume_queue" do
+      task_id = cloud_volume.attach_volume_queue(user.userid, ems.id, disk.id)
```

Comment:
I don't think ems.id is correct here, it should be a vm's ems_ref

Reply:
Alright, I think I need some clarification for how/where a VM fits into this.

Reply:
A cloud volume is being attached to a VM/instance

Reply:
@agrare ok, updated.